src/Entity/EntityIdValue.php

```diff
@@ -139,6 +138,8 @@ public function getArrayValue() {
 	 *
 	 * @param mixed $data
 	 *
+	 * @deprecated since 7.1, use callbacks in DataValueDeserializer instead
```

Comment:
It's related in that it specifies what needs to be done in order to support custom IDs.